File: python/08_070.py
def maxHolidayJoy(joyRatings, minCards, maxCards):
  n = len(joyRatings)
  if n < minCards:
     return 0
  
  # Calculate sum of initial minimum window
  current_sum = sum(joyRatings[:minCards])
  max_joy = current_sum
  
  # Try each valid window size from minCards to maxCards
  for window_size in range(minCards, min(maxCards + 1, n + 1)):
     # If increasing window size from previous iteration
     if window_size > minCards:
        current_sum += joyRatings[window_size - 1]
     
     # Initial window of current size
     temp_max = current_sum
     
     # Slide the window across the array
     for i in range(window_size, n):
        current_sum += joyRatings[i] - joyRatings[i - window_size]
        temp_max = max(temp_max, current_sum)
     
     max_joy = max(max_joy, temp_max)
     
     # Reset current_sum for next window size
     current_sum = sum(joyRatings[:window_size])
  
  return max_joy

File: data structures and algorithms/cpp/08_070.cpp
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
   // Find maximum joy from holiday cards
   int maxHolidayJoy(vector<int>& joyRatings, 
                    int minCards, int maxCards) {
      int n = joyRatings.size();
      // Check if we have enough cards
      if (n < minCards) {
         return 0;
      }
      
      // Calculate sum of initial minimum window
      int currentSum = 0;
      for (int i = 0; i < minCards; i++) {
         currentSum += joyRatings[i];
      }
      int maxJoy = currentSum;
      
      // Try each valid window size
      for (int windowSize = minCards; 
           windowSize <= min(maxCards, n); windowSize++) {
         // If increasing window size from previous iteration
         if (windowSize > minCards) {
            currentSum += joyRatings[windowSize - 1];
         }
         
         // Initial window of current size
         int tempMax = currentSum;
         
         // Slide the window across the array
         for (int i = windowSize; i < n; i++) {
            currentSum += joyRatings[i] - 
                         joyRatings[i - windowSize];
            tempMax = max(tempMax, currentSum);
         }
         
         maxJoy = max(maxJoy, tempMax);
         
         // Reset currentSum for next window size
         currentSum = 0;
         for (int i = 0; i < windowSize; i++) {
            currentSum += joyRatings[i];
         }
      }
      
      return maxJoy;
   }
};